src/components/project/project.rules.ts

```diff
@@ -1000,7 +997,10 @@ export class ProjectRules {
     id: ID,
     steps: ProjectStep[]
   ): Promise<ProjectStep> {
-    const prevSteps = await this.getPreviousSteps(id);
+    const prevSteps = [
+      ...(await this.getPreviousSteps(id)),
+      await this.getCurrentStep(id),
```

Comment:
I still think this should be reverted. We are looking for previous steps, not the current one.